### PR TITLE
[WIP] Docker image

### PR DESCRIPTION
**Pull request by zacoppotamus**
It seems like the scheduler is not running correct — I can't find Orion in the admin panel.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+ARG DB_HOST=localhost
+ARG DB_PORT=5432
+ARG DB_NAME=postgres
+ARG DB_USER=postgres
+ARG DB_PASS
+
+ENV orion_prod=postgres+psycopg2://${DB_USER}:${DB_PASS}@${DB_HOST}:${DB_PORT}/${DB_NAME}
+ENV orion_test=postgres+psycopg2://${DB_USER}:${DB_PASS}@${DB_HOST}:${DB_PORT}/postgres
+ENV airflow_db=postgres+psycopg2://${DB_USER}:${DB_PASS}@${DB_HOST}:${DB_PORT}/airflow
```
Comment (kstathou) [inline Dockerfile:16]: Added a separate db for airflow. I had to create it on AWS first. This is also passed on to `AIRFLOW__CORE__SQL_ALCHEMY_CONN`.

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+WORKDIR /airflow
+
+COPY --from=builder /install /usr/local
+COPY . ./
+COPY entrypoint.sh /entrypoint.sh
+COPY boto.cfg /etc/boto.cfg
+
+
+RUN echo $orion_prod \
```
Comment (kstathou) [inline Dockerfile]: What's the purpose of `echo $orion_prod`? It reveals AWS secrets so we should delete it if it's not needed.

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+ARG DB_PASS
+
+ENV orion_prod=postgres+psycopg2://${DB_USER}:${DB_PASS}@${DB_HOST}:${DB_PORT}/${DB_NAME}
+ENV orion_test=postgres+psycopg2://${DB_USER}:${DB_PASS}@${DB_HOST}:${DB_PORT}/postgres
+ENV airflow_db=postgres+psycopg2://${DB_USER}:${DB_PASS}@${DB_HOST}:${DB_PORT}/airflow
+
+ENV AIRFLOW_HOME /airflow/orion/core
+ENV AIRFLOW__CORE__SQL_ALCHEMY_CONN ${airflow_db}
+ENV AIRFLOW__CORE__EXECUTOR LocalExecutor
+ENV AIRFLOW__CORE__DAGS_FOLDER /airflow/orion/core/dags
```
Comment (kstathou) [inline Dockerfile:21]: Added an environmental variable pointing to the `dags` directory. Without it, airflow cannot load DAGs.

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+  && echo $DB_PASS \
+  && echo $AIRFLOW_HOME \
+  && pip install -e . \
+  && airflow initdb
+
+EXPOSE 8080
+
+ENTRYPOINT ["/entrypoint.sh"]
+CMD ["airflow", "webserver", "-D"]
+CMD ["airflow", "scheduler", "-D"]
```
Comment (kstathou) [inline Dockerfile]: Added a command to run the scheduler too.

File: orion/core/dags/orion.py [hunk 5]
```diff
@@ -1,3 +1,4 @@
+import torch
```
Comment (kstathou) [inline orion/core/dags/orion.py:1]: Solved the dependency issues by importing torch at the beginning of the script. Various solutions discussed [here](https://github.com/pytorch/pytorch/issues/2575).

File: entrypoint.sh [hunk 6]
```diff
@@ -0,0 +1,9 @@
+#!/usr/bin/env bash
+
+# Load DAGs examples (default: Yes)
+if [[ -z "$AIRFLOW__CORE__LOAD_EXAMPLES" && "${LOAD_EX:=n}" == n ]]; then
+  AIRFLOW__CORE__LOAD_EXAMPLES=False
+fi
+
+airflow scheduler &
+exec airflow webserver
```
Comment (kstathou) [inline entrypoint.sh]: What is this doing and does it differ from the `CMD` bits in the Dockerfile? (bash newbie here)

**Review by kstathou — COMMENTED** [review]
Hi @zacoppotamus! I made a few changes, airflow (with torch and the rest of the dependencies) works now! I left you some comments, let me know if the docker image works for you too and then you can merge!

File: Dockerfile (hunk 2 above)
Comment (zacoppotamus) [reply Dockerfile]: Forgot it there — was for debugging purposes :)

File: entrypoint.sh (hunk 6 above)
Comment (zacoppotamus) [reply entrypoint.sh]: Since we need two concurrently running commands, the `&` runs the first command in the background, while the `exec` replaces the currently executing bash shell with the `airflow webserver` process. This theoretically basically replicates what 
```
CMD ["airflow", "webserver", "-D"]
CMD ["airflow", "scheduler", "-D"]
```
do, so we should only use one or the other. The advantage of using an `ENTRYPOINT` command is that we can extend it at a later point if needed.

Comment (kstathou) [reply entrypoint.sh]: Great, thanks!
